cpa-repo/src/main/kotlin/no/nav/emottak/cpa/Routes.kt

```diff
@@ -180,7 +182,10 @@ fun Route.validateCpa(cpaRepository: CPARepository) = post("/cpa/validate/{$CONT
         val cpa = cpaRepository.findCpa(validateRequest.cpaId)
             ?: throw NotFoundException("Fant ikke CPA (${validateRequest.cpaId})")
         cpa.validate(validateRequest) // Delivery Failure
-        val partyInfo = cpa.getPartyInfoByTypeAndID(validateRequest.addressing.from.partyId) // Delivery Failure
+        val partyInfo = when (validateRequest.direction) {
+            IN -> cpa.getPartyInfoByTypeAndID(validateRequest.addressing.from.partyId) // Delivery Failure
+            OUT -> cpa.getPartyInfoByTypeAndID(validateRequest.addressing.to.partyId) // Delivery Failure
+        }
         val encryptionCertificate = partyInfo.getCertificateForEncryption() // Security Failure
         val signingCertificate = partyInfo.getCertificateForSignatureValidation(
```

Comment:
Lengre nede så brukes samme partyInfo objekt for å hente signingCertificate.
Når vi mottar så er det "from" partyId som skal brukes for å validere signaturen.
Når vi sender så er det "from" partiId som skal brukes for å lage signaturen.

Reply:
Good catch! Fix er pusha